chunky/src/java/se/llbit/chunky/entity/BeaconBeam.java

```diff
@@ -0,0 +1,137 @@
+import se.llbit.chunky.world.Material;
+import se.llbit.chunky.world.material.TextureMaterial;
+import se.llbit.json.JsonObject;
+import se.llbit.json.JsonValue;
+import se.llbit.math.Quad;
+import se.llbit.math.Transform;
+import se.llbit.math.Vector3;
+import se.llbit.math.Vector4;
+import se.llbit.math.primitive.Primitive;
+import se.llbit.util.JsonUtil;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+public class BeaconBeam extends Entity implements Poseable {
+
+  public static final Material beaconBeamMaterial = new TextureMaterial(Texture.beaconBeam);
+
+  private static final Quad[] beam = new Quad[]{
+      new Quad(
+          new Vector3(5 / 16.0, 0.0, 5 / 16.0),
+          new Vector3(5 / 16.0, 0.0, 11 / 16.0),
+          new Vector3(5 / 16.0, 1.0, 5 / 16.0),
+          new Vector4(1.0, 0.0, 1.0, 0.0)
+      ),
+      new Quad(
+          new Vector3(11 / 16.0, 0.0, 11 / 16.0),
+          new Vector3(11 / 16.0, 0.0, 5 / 16.0),
+          new Vector3(11 / 16.0, 1.0, 11 / 16.0),
+          new Vector4(1.0, 0.0, 1.0, 0.0)
+      ),
+      new Quad(
+          new Vector3(5 / 16.0, 1.0, 5 / 16.0),
+          new Vector3(11 / 16.0, 1.0, 5 / 16.0),
+          new Vector3(5 / 16.0, 0.0, 5 / 16.0),
+          new Vector4(1.0, 0.0, 1.0, 0.0)
+      ),
+      new Quad(
+          new Vector3(11 / 16.0, 1.0, 11 / 16.0),
+          new Vector3(5 / 16.0, 1.0, 11 / 16.0),
+          new Vector3(11 / 16.0, 0.0, 11 / 16.0),
+          new Vector4(1.0, 0.0, 1.0, 0.0)
+      )
+  };
+
+  private final JsonObject pose;
+  private double scale = 1;
+  private int height = 256;
+
+  public BeaconBeam(Vector3 position) {
+    super(position);
+    this.pose = new JsonObject();
+    pose.add("all", JsonUtil.vec3ToJson(new Vector3(0, 0, 0)));
+  }
+
+  public BeaconBeam(JsonObject json) {
+    super(JsonUtil.vec3FromJsonObject(json.get("position")));
+    this.scale = json.get("scale").asDouble(1);
+    this.height = json.get("height").asInt(256);
+    this.pose = json.get("pose").object();
+  }
+
+  @Override
+  public Collection<Primitive> primitives(Vector3 offset) {
+    Vector3 allPose = JsonUtil.vec3FromJsonArray(this.pose.get("all"));
+    return primitives(Transform.NONE
+        .translate(-0.5, -0.5, -0.5)
+        .scale(scale)
+        .rotateY(allPose.y) // Ignore x and z rotation because it breaks the beam into segments
+        .translate(0.5, 0.5, 0.5)
+        .translate(position.x + offset.x, position.y + offset.y, position.z + offset.z));
+  }
+
+  public Collection<Primitive> primitives(Transform transform) {
+    ArrayList<Primitive> faces = new ArrayList<>();
+    //Have 1 block tall model and repeat it for height * scale.
+    //This addresses the texture stretching problem and
+    //allows for the height to be changed.
+    for (int i = 0; i < height; i++) {
+      for (Quad quad : beam) {
+        quad.addTriangles(faces, beaconBeamMaterial, transform.translate(0.0, 1.0 * i * scale, 0.0));
```

Comment:
```
                -Math.cos(allPose.z - Math.PI / 2) * i * scale,
                (Math.sin(allPose.z - Math.PI / 2) * Math.sin(allPose.x - Math.PI / 2)) * i * scale,
                Math.cos(allPose.x - Math.PI / 2) * i * scale
```

I'm so fucking done with this janky rotation... For the most part the code, which Boss provided to me, was correct however there was the issue of a missing allPose.z aspect to calculating the Y offset. After adding this component the rotation along two singular axis is correct. The main issue arises from when we try and rotate along two axis where the segments break apart due to incorrect segment rotation. Additionally the order we enter rotational components into the UI can impact how the (segment) rotation is applied and in some cases segment rotation in an axis gets locked out. If someone could look at fixing the segment rotation order that might just let us rotate beams freely.

Pitch 90
![image](https://user-images.githubusercontent.com/3740917/104061059-c3dce780-51ef-11eb-95c9-cf3e241d2657.png)

Roll 45
![image](https://user-images.githubusercontent.com/3740917/104061095-db1bd500-51ef-11eb-9fbc-3aed5d5b9d12.png)


Roll 90
![image](https://user-images.githubusercontent.com/3740917/104061074-cccdb900-51ef-11eb-9f1f-43d06300b92a.png)

Pitch & Roll @ 90 
![image](https://user-images.githubusercontent.com/3740917/104061024-b7588f00-51ef-11eb-82a1-abf997a1e4bb.png)